### ISSUE TITLE
Integration test: testTLSAutoRegeneration not correctly cleaning up context

### ISSUE THREAD

**Issue by johng** [lightningnetwork/lnd#3210]
### Background

Adding additional integration test after `testTLSAutoRegeneration` results in a panic

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xc37646]
```

### Your environment

* master branch
* 4.18.0-21-generic #22-Ubuntu + darwin

**Comment by wpaulino** [comment]
Seems like there's a race condition between creating the new certificate in `testTLSAutoRegeneration` and connecting to the RPC when restarting the node. FWIW, this test should really just be a unit test.

**Comment by halseth** [comment]
Will be fixed by https://github.com/lightningnetwork/lnd/pull/3237

**Comment by wpaulino** [comment]
Fixed by #3237.
